assert.py

```diff
@@ -1,16 +1,97 @@
 import sys
 import os
 
-env_file = os.getenv('GITHUB_ENV')
-
-input_command_line = sys.argv[1]
-input_assert_file_path = sys.argv[2]
-input_specific_line = sys.argv[3]
-input_contains = sys.argv[4]
-input_expected_result = sys.argv[5]
-
-print(f"COMMAND LINE: {input_command_line}")
-print(f"ASSERT FILE PATH: {input_assert_file_path}")
-print(f"SPECIFIC LINE: {input_specific_line}")
-print(f"CONTAINS: {input_contains}")
-print(f"EXPECTED RESULT: {input_expected_result}")
+
+ENV_FILE = os.getenv('GITHUB_ENV')
+
+
+def main():
+	# This file should only run in the following conditions:
+	# It should receive exactly 6 arguments (including the file path).
+	if len(sys.argv) != 6:
+		sys.exit(
+			f"Invalid usage: not enough arguments. \"Empty\" arguments should receive \"_\".\n\
+			\rExpected:\n{sys.argv[0]} \"<CommandLine>\" \"_<AssertFilePath>\" \"_<SpecificLineNumber>\" \"_<Contains>\" \"<ExpectedResult>\"\n"
+		)
+	# The third, fourth and fifth argument should always start with "_", which will
+	# be ignored and removed from the string later.
+	if not ((sys.argv[2][0] == '_') and (sys.argv[3][0] == '_') and (sys.argv[4][0] == '_')):
+		sys.exit(
+			f"Invalid usage: arguments in the middle should have a \"_\" before the actual input.\n\
+			\rExpected:\n{sys.argv[0]} \"<CommandLine>\" \"_<AssertFilePath>\" \"_<SpecificLineNumber>\" \"_<Contains>\" \"<ExpectedResult>\""
```

Comment:
Could you explain to me why those inputs should start with `_`? I'm asking because you're replacing the `_` afterwards, therefore it doesn't seem necessary to me.

Reply:
Yeah, it's intentionally redundant. As explained above, as some inputs aren't mandatory, they become empty strings in the script call. Those empty strings seem to be ignored by python (or sys.argv I'm not sure).

<hr>
Ex.:

```
python ./assert.py "echo Hi" "" "" "Hi" "PASSED"
sys.argv should be: ["./assert.py", "echo Hi", "", "", "Hi", "PASSED"]
But it actually is: ["./assert.py", "echo Hi", "Hi", "PASSED"]
```

```
python ./assert.py "echo Hi" "./hi.txt" "1" "" "PASSED"
sys.argv should be: ["./assert.py", "echo Hi", "./hi.txt", "1", "", "PASSED"]
But it actually is: ["./assert.py", "echo Hi", "./hi.txt", "1", "PASSED"]
```
<hr>

Note that in that in the first situation sys.argv[2] is the input for "contains" and in the second situation sys.argv[2] is the input for assert_file_path.

For this somewhat small amount of inputs it's not too hard to figure out whether sys.argv[2] is a file path or a normal text. "PASSED" is always at the end of the list which is also easy to find. But what if the user's program outputs a path and the user wants to know if the output "contains" that path? This script would then try to read from a path that doesn't exist. (I did think up some other little bugs when I was writing the logic for this but I already forgot most of them).

That's why I thought it would be easier and more straightforward to add this redundant character. The user doesn't input it, it's the workflow itself that adds it and the script removes it. It just ensures every argument exist.

```
The workflow calls:  python ./assert.py "$command_line" "_$assert_file_path" "_$specific_line" "_$contains" "PASSED"
Which may turn into: python ./assert.py "echo Hi" "_" "_" "_Hi" "PASSED"

sys.argv becomes: ["./assert.py", "echo Hi", "_", "_", "_Hi", "PASSED"]
And then I process it to become:  ["./assert.py", "echo Hi", "", "", "Hi", "PASSED"]
```

Reply:
I was testing this behavior in C and it's the same. If you try to pass command line inputs as _empty strings_, they are ignored (basically the same behavior described above).